TESTS/mbed_drivers/lp_timer/main.cpp

```diff
@@ -0,0 +1,410 @@
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include "rtos.h"
+#include "hal/us_ticker_api.h"
+
+using namespace utest::v1;
+
+#define US_PER_SEC 1000000
+
+/* This test is created based on the test for Timer class.
+ * Since low power timer is less accurate than regular
+ * timer we need to adjust delta.
+ */
+
+/* NOTE: NRF51_DK platform has 32kHz timer clock which
+ *       gives resolution of 31.25us/clock tick.
+ *       Therefore for this board we need a huge delta
+ *       for the timing tests. */
+
+/* Macro to define delta based on system clock frequency.
+ * Assume that min. delta value is equal to 200 us.
+ *
+ * Calculate number of us per system tick. Then multiple calculated value by
+ * 10 and add 200. Truncate calculated delta to decimal value by casting to int.
+ *
+ * For 1 MHz clock delta is equal to 210 us.
+ *
+ * EXAMPLE K64F
+ * system clock frequency = 120 000 000 Hz (120 MHz) => 1 tick = ~0.083 us.
+ * delta = truncate(0.083 * 10 + 200) = truncate(200.83) = 200 us.
+ *
+ * EXAMPLE NRF51_DK
+ * system clock frequency = 32 678 Hz (~32 kHz) => 1 tick = 31.25 us.
+ * delta = truncate(31.25 * 10 + 200) = truncate(512.5) = 512 us.
+ */
+static const int delta_sys_clk_us = ((int) ((1.0f / ((float) osKernelGetSysTimerFreq() / US_PER_SEC)) * 10 + 200));
+
+#define DELTA_US delta_sys_clk_us
+#define DELTA_S  ((float)delta_sys_clk_us/US_PER_SEC)
+#define DELTA_MS 1
+
+/* This test verifies if low power timer is stopped after
+ * creation.
+ *
+ * Given Timer has been successfully created.
+ * When read of timer elapsed time is requested.
+ * Then result is always 0.
+ */
+void test_lptimer_creation()
```

Comment:
I would add another tests showing that stop actually works.

Reply:
Verification if stop() works correctly is provided in test_lptimer_time_accumulation() test function. This test case proves that timer counts time only between start and stop. Additional delay is provided when timer is stopped and it has been verified that this time is not counted by the timer.